runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,411 @@
+
+                        if (clientDestinations.getDestinationIdsList().contains(destination)) {
+                            log.debug("[{}] {}", clientName, "Destination already exists.");
+                            return null;
+                        }
+
+                        clientDestinations = clientDestinations.toBuilder().addDestinationIds(destination).build();
+                    } else {
+                        clientDestinations = DestinationInfoVal.newBuilder().addDestinationIds(destination).build();
+                    }
+
+                    txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinations, null);
+                    txn.commit();
+                } catch (TransactionAbortedException tae) {
+                    log.error("[{}] {}: {}", clientName, "Unable to add destination, will be retried", tae);
+                    throw new RetryNeededException();
+                }
+                return null;
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("[{}] {}: {}", clientName, "Unable to add destination", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Add a list destination to LogReplicationModelMetadataTable for monitoring
+     *
+     * @param logicalGroup String representative of the group of tables to be replicated
+     * @param remoteDestinations Clusters which the logicalGroup should be replicated to
+     */
+    public void addDestination(String logicalGroup, List<String> remoteDestinations) {
+        Preconditions.checkArgument(isValid(logicalGroup), String.format("[%s] %s", clientName, "logicalGroup is null or empty."));
+        Preconditions.checkArgument(isValid(remoteDestinations), String.format("[%s] %s", clientName, "remoteDestinations is null or empty."));
+        Preconditions.checkArgument(hasNoNullOrEmptyElements(remoteDestinations), String.format("[%s] %s", clientName, "remoteDestinations contains null or empty elements."));
+        List<String> finalRemoteDestinations = deduplicate(remoteDestinations);
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = ClientDestinationInfoKey.newBuilder()
+                            .setClientName(clientName)
+                            .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                            .setGroupName(logicalGroup)
+                            .build();
+
+                    DestinationInfoVal clientDestinations;
+
+                    if (txn.isExists(LR_MODEL_METADATA_TABLE_NAME, clientInfoKey)) {
+                        clientDestinations = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+
+                        List<String> newDestinations = new ArrayList<>(finalRemoteDestinations);
```

Comment:
why not continue using finalRemoteDestinations instead of creating another list?